x/valset-pref/msg_server_test.go

```diff
@@ -367,7 +367,7 @@ func (s *KeeperTestSuite) TestUnDelegateFromValidatorSet() {
 			delegator:      sdk.AccAddress([]byte("addr6---------------")),
 			coinToStake:    sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100_000_000)),
 			coinToUnStake:  sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(87_461_351)),
-			expectedShares: []sdk.Dec{sdk.NewDec(2_507_730), sdk.NewDec(4_137_755), sdk.NewDec(1_504_638), sdk.NewDec(4_388_526)}, // validatorDelegatedShares - (weight * coinToUnstake), for ex: 20_000_000 - (0.2 * 87_461_351)
+			expectedShares: []sdk.Dec{sdk.NewDec(2_507_730), sdk.NewDec(4_137_753), sdk.NewDec(1_504_638), sdk.NewDec(4_388_528)}, // validatorDelegatedShares - (weight * coinToUnstake), for ex: 20_000_000 - (0.2 * 87_461_351)
```

Comment:
refactored the test to check for undelegated tokens:https://github.com/osmosis-labs/osmosis/pull/5967/files#diff-e8d94c443b26e1cda354d150b30ce5542344099e0bbbb8e84e507d9df791daf7R309

however, the reason why this was changed was because we sort the tokens and take out the remaining amount from the last validator in the new algorithm